common:
    lid_topic :  "/hesai/pandar"
    imu_topic :  "/imu/data_raw"
    gnss_topic:  "/gps/gps"
    time_sync_en: false         # ONLY turn on when external time synchronization is really not possible
    time_offset_lidar_to_imu: 0.0 # 440 # 429.55  # 410 # 0.0 # Time offset between lidar and IMU calibrated by other algorithms, e.g. LI-Init (can be found in README).
                                  # This param will take effect no matter what time_sync_en is. So if the time offset is not known exactly, please set as 0.0

    # lidar_tod:  440 #time of the day of the lidar relative to start of gnss 
    # lidar_tod:  7409  # lieksa - take gps_week_number: 2367 and gps_week_milliseconds: 124840500  from inspa - compare with first ppk time
    lidar_tod: 1460      # Ranta-ahontielta
    

preprocess:
    lidar_type: 0                # 0 for hesai, 1 for Velodyne, 2 for ouster,
    point_step : 1               # i+= point_step  (16:1),(32:2),(64:4),(128:8)
    min_dist: 3 # .5 # 3
    max_dist: 100

mapping:
    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    fov_degree:    180
    det_range:   150.0
    extrinsic_est_en: true #  false      # true: enable the online estimation of IMU-LiDAR extrinsic
    
    GNSS_IMU_calibration_distance: 25 # 13 # 11
    #transform the point cloud to imu frame,  P_lidar seen from IMU will be P_imu = R*P_lidar+T

    extrinsic_R2: [0, -1, 0,   #same as rotation 90 deg around z axis 
                   1, 1,  0,
                   0, 0,  1]
    #R @ Rx(30)
    extrinsic_R: [ 0.    ,    -0.8660254,  0.5   ,   
                1.    ,     0. ,        0.   ,    
                0.  ,       0.5   ,     0.8660254]
    extrinsic_T: [-0.13, -0.061, 0] 


    # extrinsic_T: [ 0.009,  0.13,  0.09 ] #given by Teemu 
    # extrinsic_R: [ 0, 1, 0,
    #               -1, 0, 0,
    #                0, 0, 1]

    
    #logic here
                   #x of the lidar should be z of the imu
                   #y of the lidar should x of th imu
                   #z of the lidar should be y of the imu 

    # [ [0, 1, 0],    [X]_L
    #   [0, 0, 1],  * [Y]_L  = [[Y]_L, [Z]_L, [X]_L]_I
    #   [1, 0, 0]]    [Z]_L



    #his should be added to antena to shift to IMU
    extrinsic_T_GNSS2IMU: [-0.009, 0.208, -0.101] #this should not be used at all  

    #the above are the extrinsins - this should be known and untouched 
    #note that the imu and lidar should be synchronised by hardware
    #also the message times are in global frame,  while point times should be relative, starting from zero to .1 s or something 

als: 
    closest_N_files: 4
    downsample: false 
    #als_path: "/media/eugeniu/T7/Evo_drone24_from_Jesse_cropped2/"   # this is reference from Jesse
    # als_path: "/media/eugeniu/T7/Lowcost_trajs/MML_5pt/CALIBRATE_2023/data_las_cropped/"   # this is the ALS from NlS
    als_path: "/media/eugeniu/T7/Lowcost_trajs/MML_5pt/Lieksa_2021/data_las_cropped/"   # this is the ALS from NlS

    # als_path: "/media/eugeniu/T7/EVO_las/"   # dense ALS

    use_gnss: false 
    #if this is not empty we'll use the postprocessed GNSS solution 
    # postprocessed_gnss_path: "/media/eugeniu/T7/Lowcost_trajs/East_CALIBRRATE/Ita-Suomi_ETRSTM35FIN.txt" # "/media/eugeniu/T7/evo-bags/GT-MLS-25-07.txt"  
    postprocessed_gnss_path: "/media/eugeniu/T7/Lowcost_trajs/Lieksa/Ranta-ahontielta/Lieksa_Kontiovaarant_20250521.txt"
    
    saveALS_NN_2_MLS: false             #MLS map update with ALS points 

publish:
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: true #false       # false: low down the points number in a global-frame point clouds scan.
    
    #for the next make sure that the code is compiled with save option true
    # save_clouds: false           #if this is true it will save the point clouds and their pose
    save_clouds_local: false    #if this is true the clouds will be in sensor frame otherwise in global frame
    #save_clouds_path: "/home/eugeniu/x_vux-georeferenced-final/_Hesai_ref/"        #put here the path to save the clouds 

    #save_clouds_path: "/home/eugeniu/z_tighly_coupled/test/"        #put here the path to save the clouds
    #poseSavePath: "/home/eugeniu/z_tighly_coupled/test/"     #where to save the trajectory

    save_clouds:  true
    save_clouds_path: "/media/eugeniu/T7/Lowcost_trajs/clouds/test/"
    poseSavePath: "/media/eugeniu/T7/Lowcost_trajs/clouds/test/"     #where to save the trajectory
    save_poses: true 

eval:
    tree_file: "/media/eugeniu/T7/_georeferenced_latest/Ref_ALS_Hesai_fused/results/Results/Stem_curves/Found_stem_tree_attributes.txt"
    
    #vux_eval_path: "/home/eugeniu/xz_final_clouds/big_cov_init/hesai3/"
    vux_eval_path: "/home/eugeniu/x_vux-georeferenced-final/x_noise.2t_2r/"
        
